### ISSUE TITLE
ag-grid-angular public api does not expose AngularFrameworkOverrides or AngularFrameworkComponentWrapper

### ISSUE THREAD

**Issue by agenaille** [ag-grid/ag-grid#4026]
**I'm submitting a ...**
```
[x] bug report
```

We are enterprise customers.

**The Problem**
AngularFrameworkOverrides and AngularFrameworkComponentWrapper are not exported on the ag-grid angular public api.
As a result, we cannot inject those providers into a component which extends AgGridAngular.

Here is a simple example of such a component: 
```
import { Component, ComponentFactoryResolver, 
    ElementRef, ViewContainerRef, ViewEncapsulation } from '@angular/core';
import { AgGridAngular, AngularFrameworkOverrides, 
    AngularFrameworkComponentWrapper } from '@ag-grid-community/angular';

@Component({
  selector: 'my-company-grid',
  template: '',
  providers: [
    AngularFrameworkOverrides,
    AngularFrameworkComponentWrapper
  ],
  ecapsulation: ViewEncapsulation.None
})
export class MyCompanyGrid extends AgGridAngular {

  constructor(elementDef: ElementRef, viewContainerRef: ViewContainerRef, 
                     angularFrameworkOverrides: AngularFrameworkOverrides, 
                     angularFrameworkComponentWrapper: AngularFrameworkComponentWrapper,
                     _componentFactoryResolver: ComponentFactoryResolver) {
      super(elementDef, viewContainerRef, angularFrameworkOverrides, 
                angularFrameworkComponentWrapper, _componentFactoryResolver);
  }
}
```

Here is a working example of the use of `<my-company-grid></my-component-grid>` in the template.
This is using version 22.1.1
https://plnkr.co/edit/fKykvb438kb8vAAK

Here is an example of the same component which fails with the following error.
https://plnkr.co/edit/O0HvN3YqRc4VY7nX 
```
Encountered undefined provider! Usually this means you have a circular dependencies (might be caused by using 'barrel' index.ts files.
	Error: Encountered undefined provider! Usually this means you have a circular dependencies (might be caused by using 'barrel' index.ts files.
```
It's failing because those above mentioned providers aren't exported in the public api. Our company makes use of extending the grid in this manner to provide common functionality across all grids. We will need a fix for this before we can upgrade.
Thanks!

**Current behavior**
Cannot import AngularFrameworkOverrides or AngularFrameworkComponentWrapper correctly in my angular app.

**Expected behavior**
`AngularFrameworkComponentWrapper` and `AngularFrameworkOverrides` should be exported in the public api since they must be passed in the AgGridAngular constructor when extending the component.

Note, they are exported here for older versions: https://github.com/ag-grid/ag-grid-angular/blame/master/main.ts
They are now not exported here for newest version: https://github.com/ag-grid/ag-grid/blob/master/grid-packages/ag-grid-angular/projects/ag-grid-angular/src/public-api.ts

* **ag-Grid version:** 23.2.0

* **Browser:**

* **Language:** Typescript 3.1.1 / Angular 7.2.6

**Comment by LouisMoore-agGrid** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by icedjello** [comment]
Hi, 

I was unable to effectively investigate this issue as the plunkers provided are dissimilar. 

To get an understanding of this issue, I tried to recreate it by taking the component.ts file from https://plnkr.co/edit/O0HvN3YqRc4VY7nX and pasting it into two versions of our Range Selection Plunker from our documentation/archive. 

https://plnkr.co/edit/dE82waqYmDZxeIY4

https://plnkr.co/edit/TmDIoBM4pjjL52tv 

With some minor fixes to module/package setup, I was able to make both work without the error. 

In order for us to effectively investigate this, we would need more information and similar plunkers where comparisons can be easily made.

Note that this issue is getting closed, but you are welcomed to create a new issue if you can provide us with similar reproducible scenarios. 

If you are an ag-Grid Enterprise Customer, please raise this in our Zendesk support system - contact info@ag-grid.com for access. This is our primary channel for Support.

**Comment by agenaille** [comment]
Neither of your examples *extends* the `<ag-grid-angular>` component. You have rendered the `<ag-grid-angular>` component verbatim into the HTML. What i'm asking you to do is author a new web component, such as `<icedjello-ag-grid-angular>` which extends the AgGridAngular component. I will edit your 2 plunkers above later tonight so the example is easier to reproduce. If I open another issue for this, it will be the third time i've opened an issue for the same problem.

**Comment by agenaille** [comment]
@icedjello 
Here is the example of version 22.1.1
https://plnkr.co/edit/s5g3f8I5gW3PhLG2

Notice I have authored `<icedjello-ag-grid-angular>` and used it in the app.component.ts template.

Here is the example for version 23.2.0, which also has the `<icedjello-ag-grid-angular>` component, but it does not work, because some of the required ag-grid-angular service providers are not on the public api.
https://plnkr.co/edit/jE6Psxo6G421uw5d

Note the error in the dev tools:
```
(SystemJS) Encountered undefined provider! Usually this means you have a circular dependencies (might be caused by using 'barrel' index.ts files.
	Error: Encountered undefined provider! Usually this means you have a circular dependencies (might be caused by using 'barrel' index.ts files.
	    at syntaxError (https://unpkg.com/@angular/compiler@5.1.3/bundles/compiler.umd.js:528:34)
	    at eval (https://unpkg.com/@angular/compiler@5.1.3/bundles/compiler.umd.js:15795:40)
	    at Array.forEach (<anonymous>)
	    at CompileMetadataResolver._getProvidersMetadata (https://unpkg.com/@angular/compiler@5.1.3/bundles/compiler.umd.js:15780:19)
	    at CompileMetadataResolver.getNonNormalizedDirectiveMetadata (https://unpkg.com/@angular/compiler@5.1.3/bundles/compiler.umd.js:15042:30)
	    at CompileMetadataResolver.loadDirectiveMetadata (https://unpkg.com/@angular/compiler@5.1.3/bundles/compiler.umd.js:14910:42)
	    at eval (https://unpkg.com/@angular/compiler@5.1.3/bundles/compiler.umd.js:34305:72)
	    at Array.forEach (<anonymous>)
	    at eval (https://unpkg.com/@angular/compiler@5.1.3/bundles/compiler.umd.js:34304:72)
	    at Array.forEach (<anonymous>)
	Error loading https://run.plnkr.co/preview/ckeaeeyip000c3b65ay3r7uj6/main.ts
```

**Comment by agenaille** [comment]
@icedjello  I believe all that has to be done to fix this is to export the `AngularFrameworkOverrides` and `AngularFrameworkComponentWrapper` in the following file:  https://github.com/ag-grid/ag-grid/blob/master/grid-packages/ag-grid-angular/projects/ag-grid-angular/src/public-api.ts

**Comment by seanlandsman** [comment]
@agenaille This has been fixed and will be available in our next release - currently scheduled for this week

**Comment by agenaille** [comment]
@seanlandsman Thanks a bunch! I appreciate you! https://github.com/ag-grid/ag-grid/issues/4042 can likely be closed as a duplicate, in that case.
